src/Symfony/Bundle/FrameworkBundle/Test/WebTestAssertionsTrait.php

```diff
@@ -183,21 +183,27 @@ public static function assertRouteSame($expectedRoute, array $parameters = [], s
             $constraint = LogicalAnd::fromConstraints($constraint, ...$constraints);
         }
 
-        self::assertThat(static::getRequest(), $constraint, $message);
+        self::assertThat(self::getRequest(), $constraint, $message);
     }
 
-    private static function getClient(): KernelBrowser
+    private static function getClient(KernelBrowser $newClient = null): ?KernelBrowser
     {
-        if (!static::$client instanceof KernelBrowser) {
-            static::fail(\sprintf('A client must be set to make assertions on it. Did you forget to call "%s::createClient"?', __CLASS__));
+        static $client;
+
+        if (0 < \func_num_args()) {
```

Comment:
What is the reason to not use `if ($newClient) {`?

Reply:
Because we want to be able to reset the state using `getClient(null)`, while `getClient()` shouldn't reset anything

Reply:
Seems fair